libmattermost.c

```diff
@@ -108,6 +108,22 @@ json_array_to_string(JsonArray *array)
 	return str;
 }
 
+// Mattermost uses unnamed arrays that json-glib parser turns into strings ...
+JsonArray *
+json_named_array_from_string(const gchar *name, const gchar *str)
+{
+	JsonArray *array = NULL;
+	JsonParser *parser = json_parser_new();
+	gchar *data = g_strconcat("{\"", name , "\":", str, "}", NULL);
+

```

Comment:
I've just tried to parse `[{\"user_id\":\"XXXXXX\",\"category\":\"direct_channel_show\",\"name\":\"YYYYYY\",\"value\":\"false\"}]` and it's working fine for me.  What version of json-glib were you not able to parse this with?
```c
	const gchar *blah = "[{\"user_id\":\"XXXXXX\",\"category\":\"direct_channel_show\",\"name\":\"YYYYYY\",\"value\":\"false\"}]";
	JsonParser *parser = json_parser_new();
	JsonArray *array = NULL;
	GError *err;

	if (json_parser_load_from_data(parser, blah, -1, &err)) {
 		array = json_node_get_array(json_parser_get_root(parser));
 	} else {
		printf("Error: %s\n", err->message);
	}
	printf("%p\n", array);
	exit(0);
```